src/test/spec/read_write_concern/document.rs

```diff
@@ -15,80 +12,108 @@ struct TestFile {
 }
 
 #[derive(Debug, Deserialize)]
-// TODO RUST-1077: remove the #[allow(dead_code)] tag and add #[serde(deny_unknown_fields)] to
-// ensure these tests are being fully run
-#[allow(dead_code)]
-#[serde(rename_all = "camelCase")]
+#[serde(rename_all = "camelCase", deny_unknown_fields)]
 struct TestCase {
     pub description: String,
     pub valid: bool,
     pub write_concern: Option<Document>,
     pub write_concern_document: Option<Document>,
     pub read_concern: Option<Document>,
     pub read_concern_document: Option<Document>,
+    pub is_server_default: Option<bool>,
     pub is_acknowledged: Option<bool>,
 }
 
-fn write_concern_from_document(write_concern_doc: Document) -> Option<WriteConcern> {
-    let mut write_concern = WriteConcern::default();
+async fn run_document_test(test_file: TestFile) {
+    for test_case in test_file.tests {
+        let description = test_case.description.as_str();
 
-    for (key, value) in write_concern_doc {
-        match (&key[..], value) {
-            ("w", Bson::Int32(i)) => {
-                write_concern.w = Some(Acknowledgment::from(i as u32));
-            }
-            ("w", Bson::String(s)) => {
-                write_concern.w = Some(Acknowledgment::from(s));
-            }
-            ("journal", Bson::Boolean(b)) => {
-                write_concern.journal = Some(b);
-            }
-            ("wtimeoutMS", Bson::Int32(i)) if i > 0 => {
-                write_concern.w_timeout = Some(Duration::from_millis(i as u64));
+        if let Some(specified_write_concern_document) = test_case.write_concern {
+            let specified_write_concern =
+                match bson::from_document::<WriteConcern>(specified_write_concern_document) {
+                    Ok(write_concern) => {
+                        let is_valid = write_concern.validate().is_ok();
+                        assert_eq!(
+                            is_valid, test_case.valid,
+                            "Write concern deserialization/validation should fail: {}",
```

Comment:
This message seems like it could be confusing for the case where the test case is expected to be valid but isn't.

Reply:
@abr-egn looks like I refactored to address this comment -- now `test_case.valid` is always expected to be true, so the failure message should work in all cases